TicketsDemo.Mongo/TicketsContext.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using System.Configuration;
+using TicketsDemo.Data.Entities;
+using System.Data.Entity;
+using System.Data.Entity.ModelConfiguration.Conventions;
+
+
+namespace TicketsDemo.Mongo
+{
+    public class TicketsContext
+    {
+        private IMongoDatabase database;
+        public TicketsContext()
+        {
+            //string connectionString = ConfigurationManager.ConnectionStrings["MongoDb"].ConnectionString;
+            string connectionString = "mongodb://localhost:27017/TicketsDemo";
+            MongoClient client = new MongoClient(connectionString);
+            var connectionDB = new MongoUrlBuilder(connectionString);
+            database = client.GetDatabase(connectionDB.DatabaseName);
+        }
+
+        public  IMongoCollection<Train> Trains 
+        { 
+            get { return database.GetCollection<Train>("Train"); }
+        }
+        public IMongoCollection<Carriage> Carriages 
+        {
+            get { return database.GetCollection<Carriage>("Carriage"); } 
```

Comment:
With the NoSQL database you can store object graphs in the collection.
So Trains can be stored with all carriages included (and every carriage will include subsequent places list).

Here you try to 'mimic' working with relational(SQL) database while working with NoSQL. 
Try to utilize the strength of particular technology rather than натягивать ужа на енота.